addons/web/models/res_users.py

```diff
@@ -7,9 +7,16 @@
 class ResUsers(models.Model):
     _inherit = "res.users"
 
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        return super(ResUsers, self.with_context(uid_is_first=True)).name_search(name, args, operator, limit)
+
     @api.model
     def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
-        user_ids = list(super()._name_search(name, domain, operator, limit, order))
+        user_ids = super()._name_search(name, domain, operator, limit, order)
```

Comment:
Alternative solution without context key: The ORM calls `_name_search` from expression.py without limit  (and without order too) and on the contrary there is always a limit (+ order) when you come from `name_search`. Then IMO, no limit (and/or no order) means that you don't need to have the uid first (=> early return). It also makes more sense to only keep the uid as first only if there is an order.

Reply:
Hmm 🤔 I guess it's a valid alternative solution. It's a shift of control for the semantic of the function from the context key to the limit/order parameters.

1) Solution with context key:
Pros: explicit control over when we want to return your `uid` as first suggestion
Cons: context pollution

2) Solution based on order/limit:
Pros: No context pollution, avoiding *yet another context key*
Cons: implicit control through order/limit -> constrains over the suggestion mechanism and the order/limit, hard coupling them together therefor not as flexible. (ex: suggesting `uid` as first without a limit)

I guess this is more an API choice, depending on the use-cases needed to be covered.

What do you think @xavierbol (I am not really sure who to ping, I think your scope makes heavy use of the feature in general).